AIPS/aip-18.md

```diff
@@ -0,0 +1,215 @@
+discussions-to: https://github.com/Adamant-im/AIPs/issues/55
+requires: 5
+extends: https://aips.adamant.im/AIPS/aip-5
+status: Draft
+type: Standards
+category: ARC
+created: 2024-03-21
+---
+
+## Simple Summary
+
+This proposal introduces a decentralized file transfer feature in ADAMANT Messenger apps.
+
+## Abstract
+
+The ability to send files within the messaging platform enhances communication by providing users with a versatile and efficient means of sharing information. Sending files allows users to exchange various types of content, including documents, images, videos, and more, directly within the chat interface. This functionality promotes collaboration, facilitates the sharing of important documents, and enriches conversations by providing context and additional information.
+
+The critical difference of file transfer in ADAMANT is sticking to a privacy approach, utilizing decentralized file storages such as [InterPlanetary File System (IPFS)](https://en.wikipedia.org/wiki/InterPlanetary_File_System) and different storage blockchains. It follows the concept of privacy and censorship resistance appropriate to ADAMANT.
+
+## Motivation
+
+A comprehensive specification is necessary to establish a consistent and standardized approach for implementing decentralized file transfer in ADAMANT Messenger apps.
+
+## Specification
+
+In the overview, decentralized file transfer includes:
+
+- Encrypting a file utilizing the same end-to-end encryption as specified in [AIP-4](https://aips.adamant.im/AIPS/aip-4)
+- Generating file ID
+- Uploading an encrypted file to a decentralized storage node
+- Sending this ID to a recipient in the ADM rich text message
+- Whenever later, a recipient decrypts the ADM messages, downloads a file (from any node of decentralized storage), and decrypts it
+
+While encrypting and receiving/decrypting steps are described in other AIPs, sending file IDs in messages requires a specification.
+
+After a client has the file ID, it sends an ADM rich message as described in [AIP-5](https://aips.adamant.im/AIPS/aip-5), containing storage and file information. The `file_id` field represents the unique identifier of a file. Other fields provide additional information such as file type, size, preview ID, file name, and nonces for encryption.
+
+## Syntax
+
+According to [AIP-5](https://aips.adamant.im/AIPS/aip-5)), field `transaction.asset.chat.message` must contain *encrypted stringified* JSON with attached file information.
+
+The structure of the file transfer object is as follows:
+
+````
+{
+  "files": [
+    {
+      "preview": {
+        "id": "String",
+        "nonce": "String",
+        "extension": "String"
+      },
+      "id": "String",
+      "mimeType": "String",
+      "extension": "String",
+      "size": Int64,
+      "name": "String",
+      "nonce": "String",
+      "resolution": [Float, Float],
+      "duration": Float
+    }
+  ],
+  "storage": { "id": "String" },
+  "comment": "String"
```

Comment:
This is a common "comment" for all the files, right? I don't see a use case here because the transaction itself already have a "message" property. It would make more sense to move the "comment" into each file object. @StanislavDevIOS

Reply:
> I don't see a use case here because the transaction itself already have a "message" property

No. 
https://aips.adamant.im/AIPS/aip-5
It's inside the `transaction.asset.chat.message`

Reply:
You are right, now I see